ext/ddtrace.c

```diff
@@ -642,7 +642,7 @@ ZEND_METHOD(DDTrace_SpanLink, fromHeaders) {
     zend_hash_copy(Z_ARR(link->property_attributes), &result.meta_tags, NULL);
 
     zend_string *propagated_tags = ddtrace_format_propagated_tags(&result.propagated_tags, &result.meta_tags);
-    zend_string *full_tracestate = ddtrace_format_tracestate(result.tracestate, result.origin, result.priority_sampling, propagated_tags, &result.tracestate_unknown_dd_keys);
+    zend_string *full_tracestate = ddtrace_format_tracestate(result.tracestate, 0, result.origin, result.priority_sampling, propagated_tags, &result.tracestate_unknown_dd_keys);
```

Comment:
Setting a value of zero here might be misleading. 

I think we have two options here:
  1) Encode the current_span_id in the tracestate. Storing the hex encoding of a span's span_id in the tracestate is redundant but it maintains consistency with the tracestate generated by distributing tracing headers.
  2) We can avoid setting `p:......` field in the link.tracestate. This will cause the tracestate generated by span link and distributed tracing headers to diverge.

I am leaning towards option 1. As long as tracestate headers generated by the datadog and otel api contain the `p:.....` field then we should be in good shape. We do not NEED to set this field in span links but I think it is best to be consistent.

```suggestion
    zend_string *full_tracestate = ddtrace_format_tracestate(result.tracestate, result.span_id, result.origin, result.priority_sampling, propagated_tags, &result.tracestate_unknown_dd_keys);
```

Reply:
I do not have a strong opinion here and defaulted to the approach doing less computations. If you prefer 2) we can do that.

Reply:
I would prefer 2 over the current approach. In this scenario no value is better than an incorrect one